GitUI/CommandsDialogs/BrowseDialog/GitStatusMonitor.cs

```diff
@@ -394,71 +333,84 @@ private void Update()
             _statusIsUpToDate = true;
             _previousUpdateTime = Environment.TickCount;
 
+            // capture a consistent state in the main thread
+            var statusCancellationToken = _statusCancellationToken;
+            IGitModule module = Module;
```

Comment:
I don't think this will save us, should a user change working directory while we doing this.
Should it be something like 
```
var workingDir = Module.WorkingDir;
ThreadHelper.JoinableTaskFactory.RunAsync(async () => { ....
    var module = new GitModule(workingDir);
    ...
});

Reply:
To be 100% sure, you need a thread boundary. A lock, or other way to prevent working directory changes between 

var workingDir = Module.WorkingDir;
AND
ThreadHelper.JoinableTaskFactory.RunAsync(async () => { ....

http://afana.me/archive/2015/07/10/memory-barriers-in-dot-net.aspx/

Reply:
IGitModule does not allow to change its WorkingDir. I see no reason why we should create a new GitModule instance here.

Reply:
Not planning any change here then